backend/storage-service/src/jvmMain/kotlin/file/services/SynchronizationService.kt

```diff
@@ -0,0 +1,205 @@
+                {
+                    setParameter("id", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    delete from storage.synchronized_folders
+                    where id = :id and user_id = :user
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun addDevice(actor: Actor, request: SynchronizationAddDeviceRequest) {
+        if (syncthing.config.devices.find { it.id == request.id } != null) {
+            throw RPCException.fromStatusCode(HttpStatusCode.BadRequest)
+        }
+
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("device", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    insert into storage.user_devices(
+                        device_id,
+                        user_id
+                    ) values (
+                        :device,
+                        :user
+                    )
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun removeDevice(actor: Actor, request: SynchronizationRemoveDeviceRequest) {
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("id", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    delete from storage.user_devices
+                    where device_id = :id and user_id = :user
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun browseDevices(actor: Actor): PageV2<SynchronizationDevice> {
+        val devices = db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("user", actor.username)
+                },
+                """
+                        select device_id
+                        from storage.user_devices
+                        where user_id = :user
+                        limit 100
+                    """
+            ).rows.map { SynchronizationDevice(it.getField(UserDevicesTable.device)) }
+        }
+
+        return PageV2(100, devices, null)
+    }
+
+    suspend fun retrieveFolder(actor: Actor, path: String): SynchronizedFolder {
+        return db.withSession { session ->
+            val folder = session.sendPreparedStatement(
+                {
+                    setParameter("user", actor.username)
+                    setParameter("path", path)
+                },
+                """
+                        select id, path, device_id
+                        from storage.synchronized_folders
+                        where user_id = :user and path = :path
+                        limit 100
```

Comment:
Update the database such that the `(user, path)` tuple is actually unique. In that case you also do not need the `limit 100` entry.